docs/models.rst

```diff
@@ -127,19 +135,29 @@ helper table that is used for the relationship.  For this helper table it
 is strongly recommended to *not* use a model but an actual table::
 
     tags = db.Table('tags',
-        db.Column('tag_id', db.Integer, db.ForeignKey('tag.id')),
-        db.Column('page_id', db.Integer, db.ForeignKey('page.id'))
+        db.Column('tag_id', db.Integer, db.ForeignKey('tag.id'), nullable=False),
+        db.Column('page_id', db.Integer, db.ForeignKey('page.id'), nullable=False)
     )
 
     class Page(db.Model):
         id = db.Column(db.Integer, primary_key=True)
-        tags = db.relationship('Tag', secondary=tags,
-            backref=db.backref('pages', lazy='dynamic'))
+        tags = db.relationship('Tag', secondary=tags, lazy='subquery',
+            backref=db.backref('pages', lazy=True))
 
     class Tag(db.Model):
         id = db.Column(db.Integer, primary_key=True)
 
-Here we configured `Page.tags` to be a list of tags once loaded because we
-don't expect too many tags per page.  The list of pages per tag
-(`Tag.pages`) however is a dynamic backref.  As mentioned above this means
-that you will get a query object back you can use to fire a select yourself.
+Here we configured `Page.tags` to be loaded immediately after loading
+a Page, but using a separate query.  This always results in two
+queries when retrieving a Page, but when querying for multiple pages
+you will not get additional queries.
+
+The list of pages for a tag on the other hand is something that's
+rarely needed - for sure not when you just get the tags e.g. for a
```

Comment:
"for sure" and "e.g." sounds awkward, perhaps break this up into two sentences. "rarely needed. For example, you don't need them when getting the tags for a specific page."